app/src/main/java/org/schabi/newpipe/util/Localization.java

```diff
@@ -226,6 +226,17 @@ public static String shortSubscriberCount(final Context context, final long subs
                 shortCount(context, subscriberCount));
     }
 
+    public static String downloadCount(final Context context, final int downloadCount) {
+        final int properCount = downloadCount + 1;
```

Comment:
please explain, why you increase the count

Reply:
In the original code, text for one download was trigger when downloadCount was 0, and for higher amounts the count was incremented. Please check this line that I've removed from original code: https://github.com/TeamNewPipe/NewPipe/pull/6930/files#diff-22be96acbd15eb74cf0cd0f1addb47989eec8f95ae278582b5fe110629d3b6ebL489. I just moved it to this new location to have only one place with increasing the count.

Reply:
That should be corrected when passing the parameter and not in the method, because you wrote a general utility method which can be used within other contexts, too, and thus potentially causing future usages of that method to display wrong download numbers.
On the other hand, I do not understand, why the `downloadDoneCount` is incremented at the end of the method. Incrementing it at the beginning would circumvent all of these problems. I think you can safely move [this last line](https://github.com/TeamNewPipe/NewPipe/pull/6930/files#diff-22be96acbd15eb74cf0cd0f1addb47989eec8f95ae278582b5fe110629d3b6ebL494) up to [here](https://github.com/TeamNewPipe/NewPipe/pull/6930/files#diff-22be96acbd15eb74cf0cd0f1addb47989eec8f95ae278582b5fe110629d3b6ebL469) and change the condition, That would be the result:

``` Java
        downloadDoneCount++;
        if (downloadDoneCount == 1) {
            downloadDoneList.append(name);
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
                downloadDoneNotification.setContentTitle(getString(R.string.app_name));
            } else {
                downloadDoneNotification.setContentTitle(null);
            }

            downloadDoneNotification.setContentText(Localization.downloadCount(this, downloadDoneCount));
            downloadDoneNotification.setStyle(new NotificationCompat.BigTextStyle()
                    .setBigContentTitle(Localization.downloadCount(this, downloadDoneCount))
                    .bigText(name)
            );
        } else {
            downloadDoneList.append('\n');
            downloadDoneList.append(name);

            downloadDoneNotification.setStyle(new NotificationCompat.BigTextStyle().bigText(downloadDoneList));
            downloadDoneNotification.setContentTitle(Localization.downloadCount(this, downloadDoneCount));
            downloadDoneNotification.setContentText(downloadDoneList);
        }

        mNotificationManager.notify(DOWNLOADS_NOTIFICATION_ID, downloadDoneNotification.build());
    }
```